util/php2po.php

```diff
@@ -1,204 +1,204 @@
+
+// search for plural info
+$lang = "";
+$lang_logic = "";
+$lang_pnum = 2;
+
+$_idx = array_search('-p', $argv);
+if ($_idx !== false) {
+	$lang_pnum = $argv[$_idx+1];
+}
+
+$infile = file($phpfile);
+foreach($infile as $l) {
+	$l = trim($l);
+	if  (startsWith($l, 'function string_plural_select_')) {
+		$lang = str_replace( 'function string_plural_select_' , '', str_replace( '($n){','', $l) );
 	}
-	
-	$infile = file($phpfile);
-	foreach($infile as $l) {
-		$l = trim($l);
-		if  (startsWith($l, 'function string_plural_select_')) {
-			$lang = str_replace( 'function string_plural_select_' , '', str_replace( '($n){','', $l) );
-		}
-		if (startsWith($l, 'return')) {
-			$lang_logic = str_replace( '$', '', trim( str_replace( 'return ' , '',  $l) , ';') );
-			break;
-		}
+	if (startsWith($l, 'return')) {
+		$lang_logic = str_replace( '$', '', trim( str_replace( 'return ' , '',  $l) , ';') );
+		break;
 	}
-	
-	echo "Language: $lang\n";
-	echo "Plural forms: $lang_pnum\n";
-	echo "Plural logic: $lang_logic;\n";
-		
-	$out .= sprintf('"Language: %s\n"', $lang) ."\n";
-	$out .= sprintf('"Plural-Forms: nplurals=%s; plural=%s;\n"', $lang_pnum, $lang_logic)  ."\n";
-	$out .= "\n";
+}
 
-	print "\nLoading base message.po...";
-	
-	// load base messages.po and extract msgids
-	$base_msgids = array();
-	$norm_base_msgids = array();
-	$base_f = file("util/messages.po") or die("No base messages.po\n");
-	$_f = 0; $_mid = ""; $_mids = array();
-	foreach( $base_f as $l) {
-		$l = trim($l);
-		//~ print $l."\n";
-		
-		if (startsWith($l, 'msgstr')) {
-			if ($_mid != '""') {
-				$base_msgids[$_mid] =  $_mids;
-				$norm_base_msgids[preg_replace(NORM_REGEXP, "", $_mid)] = $_mid;
-				//~ print "\t\t\t".$_mid. print_r($base_msgids[$_mid], true);
-			}
-			
-			$_f = 0;
-			$_mid = "";
-			$_mids = array();
-			
-		}
-		
-		if (startsWith($l, '"') && $_f==2) {
-			$_mids[count($_mids)-1] .= "\n".$l;
-			//~ print "\t\t+mids: ".print_t($_mids, true);
-		}
-		if (startsWith($l, 'msgid_plural ')) {
-			$_f = 2;
-			$_mids[] = str_replace('msgid_plural ', '' , $l);
-			//~ print "\t\t mids: ".print_r($_mids, true);
-		}
-		
-		if (startsWith($l, '"') && $_f==1) {
-			$_mid .= "\n".$l;
-			$_mids[count($_mids)-1] .= "\n".$l;
-			//~ print "\t+mid: $_mid \n";
-		}
-		if (startsWith($l, 'msgid ')) {
-			$_f = 1;
-			$_mid = str_replace('msgid ', '' , $l);
-				$_mids = array($_mid);
-			//~ print "\t mid: $_mid \n";
+echo "Language: $lang\n";
+echo "Plural forms: $lang_pnum\n";
+echo "Plural logic: $lang_logic;\n";
+
+$out .= sprintf('"Language: %s\n"', $lang) ."\n";
+$out .= sprintf('"Plural-Forms: nplurals=%s; plural=%s;\n"', $lang_pnum, $lang_logic)  ."\n";
+$out .= "\n";
+
+print "\nLoading base message.po...";
+
+// load base messages.po and extract msgids
+$base_msgids = array();
+$norm_base_msgids = array();
+$base_f = file("util/messages.po") or die("No base messages.po\n");
+$_f = 0; $_mid = ""; $_mids = array();
+foreach( $base_f as $l) {
```

Comment:
Your guess is correct. I just averted my sensible eyes when the file was too big.